### PR TITLE
*: rename RUST_VERSION to rust-toolchain

### PR DESCRIPTION
**Pull request by overvenus**
## What have you changed? (mandatory)

Rename RUST_VERSION to rust-toolchain.

## What are the type of the changes? (mandatory)

- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

No.

## Does this PR affect documentation (docs) update? (mandatory)

No.

## Does this PR affect tidb-ansible update? (mandatory)

No.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -42,16 +42,16 @@ jobs:
...
           name: Setting up Rust and the rest of the Build Environment
           command: |
-            RUST_VERSION=`tail -n 1 RUST_VERSION`
+            rust-toolchain=`tail -n 1 rust-toolchain`
             if [[ ! -e $HOME/.cargo ]]; then
               curl https://sh.rustup.rs -sSf |
-                sh -s -- --no-modify-path --default-toolchain ${RUST_VERSION} -y;
-              rustup default ${RUST_VERSION};
+                sh -s -- --no-modify-path --default-toolchain ${rust-toolchain} -y;
+              rustup default ${rust-toolchain};
```
Comment (BusyJay) [inline .circleci/config.yml]: Seems it's not necessary anymore.

File: rust-toolchain [hunk 2]
```diff
@@ -0,0 +1 @@
+nightly-2018-04-06
```
Comment (siddontang) [inline rust-toolchain]: seem we need to upgrade this now. 

PTAL @brson

File: .circleci/config.yml (hunk 1 above)
Comment (siddontang) [reply .circleci/config.yml]: /cc @brson

File: rust-toolchain (hunk 2 above)
Comment (overvenus) [reply rust-toolchain]: It's in the release-2.0 branch, I think we should use the same toolchain as before.

**Pull request merged**